### ISSUE TITLE
Unable to add custom pod/deployment/statefullset labels or annotations

### ISSUE THREAD

**Issue by gwvandesteeg** [neo4j-contrib/neo4j-helm#47]
There is no functionality in the chart to add your own custom annotations/labels at either the core or replica sets, in either the pods, the deployments, or the statefulset.  However they can be added to the service definitions, so they appear in some parts, but not all.

Add ability to specify:
- annotations
- labels

To the:
- pods
- deployments
- replicaset/statefullset

**Comment by moxious** [comment]
I can see a clear need for this one, but please help me out and provide a bit more information.  We've previously accepted PRs that do allow you to do this (add custom annotations) for the service and the discovery service.

Now, there are a lot of different resources in the helm chart, and theoretically people may have good reasons for custom annotating any one of them.  So this is a potential explosion of extra config parameters to allow custom annotation.

On the other hand -- it's also easy to helm install followed by kubectl patch to add whatever annotations you want.  And so even if the helm chart itself doesn't support these kinds of custom annotations, it's still possible.

Bottom line -- I see the need.   Have you considered kubectl patch?  Are there reasons why this isn't a good solution for your case?   And if it isn't a good solution - do you use other helm charts we might look at to see what a convenient pattern might look like?

**Comment by gwvandesteeg** [comment]
We use a GitOps (fluxcd).

In a GitOps environment you cannot modify (shouldn't alter) the deployed resources after they've been deployed as such without it going through the entire deployment pipeline, and if you're deploying the Helm charts directly all information such as labels and annotations must be passed in during the deployment of the resources as per the HelmRelease CRD.

**Comment by moxious** [comment]
@gwvandesteeg suppose I just added a base annotations and labels parameter, let you specify it *once*, and applied it all over the place to all resources.   Would this fit the bill, or do you need the ability to selectively apply certain annotations & labels on certain resource types?  Because that sounds quite complex and intricate to be able to reasonably support & test.

**Comment by gwvandesteeg** [comment]
Once is enough, i've got a sample up for something that'll do for now in #55

**Comment by moxious** [comment]
Thanks for the PR - this is extensible enough that we can use the same technique on other resources later if needed.  Merged.
